flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/catalog/hive/HiveCatalogHiveMetadataTest.java

```diff
@@ -240,6 +246,84 @@ void testAlterPartitionColumnStatistics() throws Exception {
+                        Column.physical("second", DataTypes.DATE()),
+                        Column.physical("third", DataTypes.INT())),
+                Collections.emptyList(),
+                null);
+    }
+
+    @Test
+    public void testGetTableAndColumnStatistics() throws Exception {
+
+        // create a test table
+        catalog.createDatabase(db1, createDb(), false);
+        final ResolvedSchema resolvedSchema = createSchemaForStatistics();
+        final CatalogTable origin =
+                CatalogTable.of(
+                        Schema.newBuilder().fromResolvedSchema(resolvedSchema).build(),
+                        TEST_COMMENT,
+                        createPartitionKeys(),
+                        getBatchTableProperties());
+        CatalogTable catalogTable = new ResolvedCatalogTable(origin, resolvedSchema);
+
+        catalog.createTable(path1, catalogTable, false);
+
+        int second = 29;
+        final int partitionFileSize = 20;
+        final String partitionName = "third";
+
+        List<Map<String, String>> partitionSpecs = new ArrayList<>();
+        Map<String, String> partitionSpec = null;
+
+        while (second-- > 10) {
+            int third = 20;
+            while (third-- > 0) {
+                partitionSpec = new HashMap<String, String>();
+                partitionSpec.put("second", "2010-04-" + second + " 00:00:00");
+                partitionSpec.put(partitionName, Integer.toString(third));
+                createPartition(partitionSpec);
+                partitionSpecs.add(partitionSpec);
+            }
+        }
+
+        final List<CatalogPartitionSpec> catalogPartitionSpecList =
+                partitionSpecs.stream()
+                        .map(p -> new CatalogPartitionSpec(p))
+                        .collect(Collectors.toList());
+
+        final List<CatalogTableStatistics> partitionStatistics =
+                catalog.bulkGetPartitionStatistics(path1, catalogPartitionSpecList);
+        assertThat(partitionStatistics.size()).isEqualTo(partitionSpecs.size());
+
+        final List<CatalogColumnStatistics> tableColumnStatistics =
+                catalog.bulkGetPartitionColumnStatistics(path1, catalogPartitionSpecList);
+
+        assertThat(tableColumnStatistics.size()).isEqualTo(catalogPartitionSpecList.size() + 1);
+
+        final CatalogColumnStatistics catalogColumnStatistics =
+                tableColumnStatistics.get(tableColumnStatistics.size() - 1);
+        CatalogColumnStatisticsDataLong columnStatisticsDataLong =
+                (CatalogColumnStatisticsDataLong)
+                        catalogColumnStatistics.getColumnStatisticsData().get(partitionName);
+
+        assertThat(columnStatisticsDataLong.getNdv()).isEqualTo(catalogPartitionSpecList.size());
+        assertThat(columnStatisticsDataLong.getMin()).isEqualTo(0);
+        assertThat(columnStatisticsDataLong.getMax()).isEqualTo(partitionFileSize - 1);
```

Comment:
other columns' statistics does not been verified

Reply:
+1